Automattic/wp-calypso
# Tidy up Image/Video embeds in Reader Streams

The image and video dimensions in the Reader streams seems a bit broken in places.

This is a follow up to https://github.com/Automattic/wp-calypso/pull/85371 which introduced support for the pocket casts embed.

This PR will try to fix the following issues;
- [x] Video embed dimensions not expanding properly after clicked - they should take up entire width and hide excerpt if necessary
- [x] Refactor the featured image code to be more readable - it has grown wild with the latest iterations
- [x] Image or video embeds will be consistent across all Reader streams
- [x] Pocket casts embed (or any embed) that can fit the container width

### Changes
* This updates the compact post to use the full width when expanding a video/pocket casts embed
* The feature image component is refactored to make it easier to understand what container dimensions and subsequent styling an image or video embed should have
* This passes the `isCompactPost` and `hasExcerpt` properties to some components that were missing them, which meant the featured asset wasn't been rendered consistently

See pe7F0s-1si-p2 for more details

### Testing
* Apply patch and go to a test stream with various types of post types - `read/feeds/7025646` is a good example
* Confirm the images and video render and behave as expected on Reader
